### PR TITLE
[Haskell] Challenge 2 (ReadyForMerge)

### PR DESCRIPTION
**Pull request by halogenandtoast**

### CODE REVIEW

**Comment by Flickdm** [comment]
Overall code was clear and concise. My knowledge of Haskell syntax is limited but the logic was clear and so was the style making it easy to read.

File: challenge_2/haskell/halogenandtoast/src/Challenge.hs [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+import Data.Maybe ( fromMaybe )
+import Data.HashMap.Strict ( alter
+                           , toList
+                           , filterWithKey
+                           , empty
+                           , HashMap(..)
+                           )
+
+counts :: [String] -> HashMap String Int
```
Comment (erocs) [inline challenge_2/haskell/halogenandtoast/src/Challenge.hs]: I would alter this to be generically applicable.
counts :: (Eq a, Hashable a) => [a] -> HashMap a Int

Comment (halogenandtoast) [inline challenge_2/haskell/halogenandtoast/src/Challenge.hs]: Good point, I did this in challenge 3 with this function.

**Comment by halogenandtoast** [comment]
Feedback addressed.

**Pull request merged**